py_zipkin/zipkin.py

```diff
@@ -201,6 +233,29 @@ def __init__(
         # 'sa' binary annotation that shows where the request is going.
         self.sa_endpoint = None
 
+        # It used to  be possible to override timestamp and duration by passing
+        # in the cs/cr or sr/ss annotations. We want to keep backward compatibilty
+        # for now, so this logic overrides self.timestamp and self.duration in the
+        # same way.
+        # This doesn't fit well with v2 spans since those annotations are gone, so
+        # we also log a deprecation warning.
+        if {'sr', 'ss'}.issubset(set(self.annotations.keys())):
```

Comment:
IMO this is a bit _too_ clever and makes it harder to read than the more verbose `'sr' in self.annotations and 'ss' in self.annotations`